platforms/quorum/charts/node_tessera/templates/deployment.yaml

```diff
@@ -195,43 +182,71 @@ spec:
-
-          printenv;
-          wget -q http://dev.mysql.com/get/Downloads/Connector-J/mysql-connector-java-8.0.25.tar.gz;
-          tar -xf mysql-connector-java-8.0.25.tar.gz;
-          cp /mysql-connector-java-8.0.25/mysql-connector-java-8.0.25.jar $QUORUM_HOME;
-          SQL_JAR=$QUORUM_HOME/mysql-connector-java-8.0.25.jar;
+          - |-
+            apk update && apk add jq curl
+            mkdir -p $TESSERA_HOME/logs;
+            mkdir -p $TESSERA_HOME/tm;
+            DDIR=$TESSERA_HOME/tm;
+            PASSWORD=$(cat $TESSERA_HOME/crypto/keys/password);
+            validateVaultResponse () {
+              if echo ${2} | grep "errors"; then
+                echo "ERROR: unable to retrieve ${1}: ${2}"
+                exit 1
+              fi
+              if  [ "$3" == "LOOKUPSECRETRESPONSE" ]
+              then
+                http_code=$(curl -sS -o /dev/null -w "%{http_code}" \
+                --header "X-Vault-Token: ${VAULT_CLIENT_TOKEN}" \
+                ${VAULT_ADDR}/v1/${vault_secret_key})
+                curl_response=$?
+                if test "$http_code" != "200" ; then
+                    echo "Http response code from Vault - $http_code"
+                    if test "$curl_response" != "0"; then
+                      echo "Error: curl command failed with error code - $curl_response"
+                      exit 1
+                    fi
+                fi
+              fi
+            }
 
-          # Extracting version from manifest incase we want to add version specific steps/config
-          TESSERA_VERSION=$(unzip -p /tessera/tessera-app.jar META-INF/MANIFEST.MF | grep Tessera-Version | cut -d\" \" -f2);
-          echo \"Tessera version (extracted from manifest file): ${TESSERA_VERSION}\";
-          
-          TESSERA_JAR=/tessera/tessera-app.jar;
-           
-          CONFIG_TMPL=$(cat ${QUORUM_HOME}/tessera-config.json.tmpl);
-          CONFIG_FINAL=$(echo $CONFIG_TMPL | sed \"s@%THIS_PRIV_KEY%@${QUORUM_HOME}/crypto/keys/tm.key@g\" |  sed \"s@%THIS_PUB_KEY%@${QUORUM_HOME}/crypto/keys/tm.pub@g\");
-          
-          echo $CONFIG_FINAL >  ${DDIR}/tessera-config-with-hosts.json;
-          cat  ${DDIR}/tessera-config-with-hosts.json;
+            KUBE_SA_TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+            echo "Getting secrets from Vault Server: ${VAULT_ADDR}"
+            # Login to Vault and so I can get an approle token
+            VAULT_CLIENT_TOKEN=$(curl -sS --request POST ${VAULT_ADDR}/v1/auth/${KUBERNETES_AUTH_PATH}/login \
+              -H "Content-Type: application/json" \
+              -d '{"role":"'"${VAULT_APP_ROLE}"'","jwt":"'"${KUBE_SA_TOKEN}"'"}' | \
+              jq -r 'if .errors then . else .auth.client_token end')
+            validateVaultResponse 'vault login token' "${VAULT_CLIENT_TOKEN}"
+            
+            # required for tessera vault integration
+            export HASHICORP_TOKEN=${VAULT_CLIENT_TOKEN};        
```

Comment:
Why not used this directly in line 218 above?

Reply:
added.